plugins/chat/assets/javascripts/discourse/components/chat-composer.js

```diff
@@ -248,6 +248,38 @@ export default class ChatComposer extends Component {
       return;
     }
 
+    // Check if the message is a reaction to the latest message in the channel.
+    const message = this.draft.message.trim();
+    let reactionCode = "";
+    if (message.startsWith("+")) {
+      const reaction = message.substring(1);
+      // First check if the message is +{emoji}
+      if (replacements[reaction]) {
+        reactionCode = replacements[reaction];
+      } else {
+        // Then check if the message is +:{emoji_code}:
+        const emojiCode = reaction.substring(1, reaction.length - 1);
+        if (emojis.includes(emojiCode)) {
+          reactionCode = emojiCode;
+        }
+      }
+    }
+
+    if (reactionCode) {
+      if (this.lastMessage?.id) {
```

Comment:
I'm wondering if there is a risk of encountering timing problem where the user intended to react to `message A` but end up reacting to `message B` if `message B` arrives after the user has hit send.

Reply:
There's a small window where that's possible, or even that `message B` arrives just before the user hits send, but close enough that they don't have the reaction time to stop themselves from pressing Enter. :D 

I think this is would be largely avoided, however, thanks to the `user is typing...` indicator, which would warn the user there's a chance they'll end up reacting to the wrong message if they send it now.